src/app.js

```diff
@@ -2,12 +2,155 @@
+    messageHash[messageType].forEach((message) => {
+      $('#status-messages ul').append($(`<li>${messageType}:  ${message}</li>`));
+    })
+  }
+  $('#status-messages').show();
+}
+
+const updateStatusMessageWith = (message) => {
+  $('#status-messages ul').empty();
+  $('#status-messages ul').append(`${message}</li>`);
+  $('#status-messages').show();
+}
+
+const events = {
+  // ------------- Add a new Trip --------------
+  addTrip(event) {
+    console.log(this);
+    event.preventDefault();
+    const tripData = {};
+    fields.forEach((field) => {
+      tripData[field.toLowerCase()] = $(`.add-trip-form input[name=${field}]`).val();
+      console.log(field);
+      console.log(tripData[field]);
+    });
+
+    const trip = new Trip(tripData);
+
+    if (trip.isValid()) {
+
+      trip.save({}, {
+        success: events.successfullSave,
+        error: events.failedSave,
+      });
+    } else {
+      updateStatusMessageFrom(trip.validationError);
+    }
+
+  },
+  successfullSave(trip, response) {
+    updateStatusMessageWith(`${trip.get('name')} added!`);
+    tripList.add(trip);
+  },
+  failedSave(trip, response) {
+    updateStatusMessageFrom(response.responseJSON.errors);
+    trip.destroy();
+  },
+  sortTrips(event) {
+    $('.current-sort-field').removeClass('current-sort-field');
+    $(this).addClass('current-sort-field');
+
+    const classes = $(this).attr('class').split(/\s+/);
 
+    classes.forEach((className) => {
+      if (fields.includes(className)) {
+        console.log(className);
+        if (className === tripList.comparator) {
+          tripList.models.reverse();
+          tripList.trigger('sort', tripList);
+        }
+        else {
+          tripList.comparator = className;
+          tripList.sort();
+        }
+      }
+    });
+  },
+}
+
+// ----------- All trips / Details ------------
 $(document).ready( () => {
-  $('main').html('<h1>Hello World!</h1>');
+  tripTemplate = _.template($('#trip-template').html());
+  tripDetails = _.template($('#trip-details-template').html());
+  $('.add-trip-form').submit(events.addTrip);
+  $('.sort').click(events.sortTrips);
+  tripList.on('sort', render, tripList);
+
+  $('#trip-list').hide();
+  tripList.on('update', render, tripList);
+  tripList.fetch();
+
+  $('#all-trips').on('click', function() {
+    $('#trip-list').toggle();
+  });
+
+// ----------- Trip Details -------------
+  $('#trip-list').on('click', 'tr', function() {
+    $('tr').css('background', '');
+    const id = $(this).attr('id');
+    const url = new Trip(this).url();
+
+    $(`#${id}`).css('background','pink');
+
+    $.get(url, function(response) {
+      $(`#trip-details`).html(tripDetails(response));
+    });
+  });
+
+  // ----------- Modal ------------
+  $('#add-trip-button').on('click', function() {
+    $('#add-trip').css('display', 'block');
+  });
+
+  $(document).on('click', function() {
+    const modal = document.getElementById('add-trip');
```

Comment:
Avoid mixing JavaScript and jQuery selections, use one or the other, in this case I'd suggest jQuery.